vms/evm/metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package prometheus
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/ava-labs/libevm/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Gatherer implements [prometheus.Gatherer] interface by
+// gathering all metrics from the given Prometheus registry.
+type Gatherer struct {
+	registry Registry
+}
+
+var _ prometheus.Gatherer = (*Gatherer)(nil)
```

Comment:
nit: Can we move this up to the top of the file (aligns with all the other type assertions in avalanchego)